roles/openshift_preflight/common/tasks/main.yml

```diff
@@ -0,0 +1,22 @@
+---
+# check content available on all hosts
+- when:
+    - not containerized
+  block:
+
+    - name: determine if yum update will work
+      action: check_yum_update
+      register: r
+
+    - set_fact:
+        oo_preflight_check_results: "{{ oo_preflight_check_results + [r|combine({'_task': 'check content available on all hosts'})] }}"
```

Comment:
Thanks for the snippets, it makes it easier to comprehend.
I find that pattern unsatisfactory to work with.

Every time a task is added or removed or moved, one needs to (unless otherwise automated with templates / YAML generation) manually update the definition of `openshift_preflight_a_error_count`. It is so easy to make a mistake an use `result_7` twice, etc. We should not ask humans to type and maintain those enumerated summations.

Similarly, every time a role is added/removed, etc, the `post_tasks` in the playbook needs to change. If I add a role `d` and forget to add `openshift_preflight_d_error_count` we have a silent bug...

Put those things together, and there is room for a complex web of bugs.


---

I didn't come up with the idea myself, but can't remember/find where I took it from... I would also prefer an explicitly documented feature. Here is what I find in the docs:

http://docs.ansible.com/ansible/playbooks_variables.html#variable-precedence-where-should-i-put-a-variable

> A lot of folks may ask about how variables override another. (...)
> Avoid defining the variable “x” in 47 places and then ask the question “which x gets used”. Why? Because that’s not Ansible’s Zen philosophy of doing things. (...)
> Figure out where to define a variable, and don’t make it complicated.

We clearly go against the philosophy here by implementing a workaround that contradicts Ansible's ideal (fail on first error).

But:

> If multiple variables of the same name are defined in different places, they get overwritten in a certain order. (...)

Doesn't that count as making the variable overriding well-defined and documented?

Variable overriding is a thing, and even the precedence order is listed in the docs (we only use `set_fact`).